blueocean-web/src/main/js/main.jsx

```diff
@@ -26,7 +26,7 @@ class App extends Component {
             <div className="Site">
                 <div id="outer">
                     <header className="global-header">
-                        <ExtensionPoint name="jenkins.logo.top"/>
+                        <Extensions.Renderer name="jenkins.logo.top"/>
```

Comment:
@kzantow yes, that is really my only suggestion, that it is called out as an extension point (ie "things plugin here") - but its terminology. IF there was a group of your working on this and it made sense to you, then disregard my comment. The concept is great. And Extensions could be imported as ExtensionPoints or something like you say, via ES6, then that is ok. 

It is a little inconsistent with the naming in the config file as you say otherwise.